Post checkout

pyepye opened the pull request:
I wrote this into my own app originally but thought it was probably helpful to everyone so made it a little more generic and move it over.

* Reusable template to easily add PaddleJS and `Paddle.Setup` to a page
* Checkout model and a post checkout view to save basic Checkout information
* Some JS to automatically send data to the above API on checkout complete
* Doc update with the above info
* Make `on_subscription_created` use `update_or_create`. This is incase someone wants to create a subscription before a Webhook is received

FPurchess commented on djpaddle/templates/djpaddle_post_checkout.html:
```diff
@@ -0,0 +1,53 @@
+}
+
+function addCheckoutCompleteToPaddleButtons() {
+    paddleButtons = document.getElementsByClassName("paddle_button");
```
`var` missing?

FPurchess commented on docs/paddle_checkout.rst:
```diff
@@ -0,0 +1,82 @@
+    post_save.connect(PaddlecheckoutReciever, sender=Checkout)
+
+
+Another option is a background task to compare each ``Checkout.id`` against each ``Subscription.checkout_id`` to ensure no Webhooks have been missed.
```
@pyepye do you think it makes sense to create a management-command for that as well?

FPurchess commented on docs/paddle_checkout.rst:
```diff
@@ -0,0 +1,82 @@
+Other post checkout options
+---------------------------
+
+If you want to manually configure what your happens after a checkout has been completed instead of using the ``checkout_push.html`` template please see:
```
`what your happens` ==> `what happens`

FPurchess commented on tests/test_checkout_api.py:
```diff
@@ -0,0 +1,86 @@
+from djpaddle.models import Checkout
+
+
+class TestWebhook(TestCase):
```
Should be `CheckoutTestCase`, no?

pyepye commented:
@FPurchess  I've fixed all those bits up thanks.

I also did some testing with Paddle's redirects and it didn't quite work as expected so I've added in a way to redirect with `djpaddle_post_checkout.html` and some info in the docs about it to try and save some people the same hassle.

I've also just made a note that a few management commands are coming soon.

FPurchess commented on djpaddle/templates/djpaddle_post_checkout.html:
```diff
@@ -19,27 +14,49 @@
+            } else if (xhr.status === 204) {
+                console.log("Checkout complete with ID: " + data.checkout.id)
+            } else {
+                alert("There was a problem with the request.");
```
@pyepye this should rather redirect to an error-page

pyepye replied:
I'm not sure it's a good idea to redirect to any error page as it's not the Paddle checkout that has failed. It's just the dj-paddle post checkout request which isn't _that_ important as a webhook should come along soon and ensure we know the checkout was a success.

FPurchess commented on djpaddle/templates/djpaddle_post_checkout.html:
```diff
@@ -19,27 +14,49 @@
+    xhr.onreadystatechange = function(){
+      // Process the server response here.
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            console.log(xhr.status);
```
@pyepye please remove `console.log` or only show them when `DEBUG=True`

FPurchess commented on docs/paddle_checkout.rst:
```diff
@@ -0,0 +1,144 @@
+    from djpaddle.models import Checkout, Subscription
+
+
+    def PaddlecheckoutReciever(sender, instance, created, **kwargs):
```
@pyepye any reason why you did not use snake_case?

FPurchess commented on djpaddle/models.py:
```diff
@@ -201,19 +201,29 @@ def __str__(self):
 
 
+class Checkout(models.Model):
+    # Backup model used to store checkout info from PaddleJS. Acts as
```
@pyepye Let's use tripple-quotes here (""") to be consistent on how to comment classes

FPurchess commented on docs/paddle_checkout.rst:
```diff
@@ -0,0 +1,144 @@
+Post-Checkout Order Information
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once the user has completed an order you should there are several ways you can then get order details.
```
@pyepye is there something missing in this sentence?

pyepye commented:
Thanks for the review again. I've cleaned up those bits.
Sorry about the poor English in the docs, you would never guess it was my first language.
